### PR TITLE
[14.0][MIG] pos_payment_terminal

### PR DESCRIPTION
**Pull request by alexis-via**
This PR works and has been tested on real Ingenico card reader (for France, with Telium) connected via pywebdriver (master branch).

It is a full re-write of the module, in order to take advantage of the new payment interface JS framework of Odoo POS.

FR translation updated.

### CODE REVIEW

File: pos_payment_terminal/static/src/js/payment_terminal.js
Comment (legalsylvain) [inline pos_payment_terminal/static/src/js/payment_terminal.js]: maybe ``<=`` here, don't you think ?

File: pos_payment_terminal/views/assets.xml
Comment (legalsylvain) [inline pos_payment_terminal/views/assets.xml]: should be removed, I guess.

**Review by legalsylvain — COMMENTED** [review]
Huge refactor ! Thanks a lot to begin to port PoS module in the new framework. 
I did a superficial review and LGTM. 
As far as I understand, we don't need now specific button to send to the terminal ? (I see you removed all the xml file).
Side question : do you like the new framework ? is it better than the previous one ?

thanks.

File: pos_payment_terminal/models/__init__.py
Comment (ivantodorovich) [inline pos_payment_terminal/models/__init__.py]: Is it required or deprecated?
if it's deprecated we should remove the file entirely

File: pos_payment_terminal/tests/__init__.py
Comment (ivantodorovich) [inline pos_payment_terminal/tests/__init__.py]: to check

File: pos_payment_terminal/models/__init__.py
Comment (alexis-via) [reply pos_payment_terminal/models/__init__.py]: I don't know. I didn't write that code; it was written by Acsone. I don't what is the purpose and how it works. That's why I just commented it and I didn't want to remove it.

**Comment by sbidoul** [comment]
@alexis-via this is not finished, right ?

This module needs to support [send_payment_cancel](https://github.com/odoo/odoo/blob/66b468626e046a13a463cff3a2285801016f3ed7/addons/point_of_sale/static/src/js/payment.js#L72) and [send_payment_request must return a promise](https://github.com/odoo/odoo/blob/66b468626e046a13a463cff3a2285801016f3ed7/addons/point_of_sale/static/src/js/payment.js#L54) that manages the transaction status via the `change:status` event and the [payment transactions list](https://github.com/akretion/pywebdriver/blob/a16d833fbcf7bbaa6370b1f61f66c7e0f38a45c9/pywebdriver/plugins/payment_base_driver.py#L59) we get in it.

Are you or @PierrickBrun working on this these days ? Shall we coordinate or efforts ?

File: pos_payment_terminal/models/__init__.py
Comment (sbidoul) [reply pos_payment_terminal/models/__init__.py]: IIRC this was done to save the payment transaction id in the database when the terminal driver returns it. It [looks like there is native support for that in the point_of_sale](https://github.com/odoo/odoo/blob/66b468626e046a13a463cff3a2285801016f3ed7/addons/point_of_sale/static/src/js/payment.js#L48-L51) module so we should probably use that.

**Comment by sbidoul** [comment]
@alexis-via I made this PR green in https://github.com/akretion/pos/pull/20 and improved it a little bit.

Now I'll work on handling the transaction statuses from the payment terminal drivers, compatible with [pywebdriver](https://github.com/akretion/pywebdriver/blob/master/pywebdriver/plugins/payment_mock_driver.py).

File: pos_payment_terminal/static/src/js/payment_terminal.js
Comment (sbidoul) [reply pos_payment_terminal/static/src/js/payment_terminal.js]: Done in https://github.com/akretion/pos/pull/20

File: pos_payment_terminal/models/__init__.py
Comment (sbidoul) [reply pos_payment_terminal/models/__init__.py]: I removed this code in https://github.com/akretion/pos/pull/20

File: pos_payment_terminal/tests/__init__.py
Comment (sbidoul) [reply pos_payment_terminal/tests/__init__.py]: also removed in https://github.com/akretion/pos/pull/20

**Review by sbidoul — APPROVED** [review]

**Comment by sbidoul** [comment]
@ivantodorovich @legalsylvain your review comments have been processed.

**Comment by sbidoul** [comment]
BTW, when the transaction has been sent to the terminal, this is what happens in the POS - I post this so others don't have to research why this is normal :)

image

This is normal behaviour until we have transaction status from the driver: the user has to manually confirm that the transaction suceeded by clicking "Force done". The "Connection error" message is hard coded in Odoo.

**Review by ivantodorovich — APPROVED** [review]
LGTM! :+1: 

Please squash migration commits before merging

**Comment by sbidoul** [comment]
> Please squash migration commits before merging

Well, there are two authors and this is more a rewrite than a migration, and the commit history helps understand why things are as they are.

**Comment by ivantodorovich** [comment]
> > Please squash migration commits before merging
> 
> Well, there are two authors and this is more a rewrite than a migration, and the commit history helps understand why things are as they are.

I'd at least clean a bit the commits, whilst keeping the authors

For instance there are commits like
- `pos_payment_terminal: remove commented lines`,
- `pos_payment_terminal: remove dead code` , 
- `pos_payment_terminal: fix lint issues`, 
- `pos_payment_terminal: black, isort and pre-commit stuff`



----

It's also a good opportunity to clean the previous history.
There are things like this that could be squashed:

image

as well as the usual administrative commits

> Squash administrative commits (if any) with the previous commit for reducing commit noise. They are named as "[UPD] README.rst", "[UPD] Update $MODULE.pot", "Update translation files" and similar names, and comes from OCA-git-bot, oca-travis or oca-transbot. IMPORTANT: Don't squash legit translation commits, authored by their translators, with the message "Translated using Weblate (...)".

https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by sbidoul** [comment]
I try to make small commits that are easy to review, and also sufficiently atomic that they are easy to revert. I also strive to not rewrite past history. Anyway,  I can't do the squashing myself here as this is not a branch I own.

**Comment by ivantodorovich** [comment]
I'm not blocking in this 👍🏻

**Pull request merged**
